### PR TITLE
[Feature] support column mode partial update

### PR DESCRIPTION
**Pull request by luohaha**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #20436

## Problem Summary(Required) ：
This is the 4th part of column mode partial update feature. It contains:
1. Segment iterator 
2. Incrememtal/full clone
3. schema change
4. Load/Update support.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/column/field.h [hunk 1]
```diff
@@ -157,6 +161,9 @@ class Field {
 
     ColumnPtr create_column() const;
 
+    void set_uid(ColumnUID uid) { _uid = uid; }
+    ColumnUID uid() const { return _uid; }
```
Comment (chaoyli) [inline be/src/column/field.h:165]: `const ColumnUID& uid() const { return _uid; }`

File: gensrc/proto/olap_file.proto [hunk 2]
```diff
@@ -88,6 +88,7 @@ enum PartialUpdateMode {
     UNKNOWN_MODE = 0;
```
Comment (chaoyli) [inline gensrc/proto/olap_file.proto:88]: UNKNOWN is enough for this name.

File: be/src/column/field.h (hunk 1 above)
Comment (luohaha) [reply be/src/column/field.h:165]: ColumnUID is `int32_t`, not need this?

File: gensrc/proto/olap_file.proto (hunk 2 above)
Comment (luohaha) [reply gensrc/proto/olap_file.proto:88]: `UNKNOWN` will be duplicate name with other enum in namespace `starrocks` , and cause compile error. So I think it is OK to use this.

File: be/src/column/field.h (hunk 1 above)
Comment (chaoyli) [reply be/src/column/field.h:165]: It's a style recommendation.

File: be/src/storage/olap_common.h [hunk 3]
```diff
@@ -269,6 +270,7 @@ struct OlapReaderStatistics {
 };
 
 typedef uint32_t ColumnId;
+typedef int32_t ColumnUID;
```
Comment (chaoyli) [inline be/src/storage/olap_common.h:273]: Will the ColumnUID be negative?

File: be/src/column/field.h [hunk 4]
```diff
@@ -176,6 +183,7 @@ class Field {
     int32_t _length = 0;
     uint8_t _short_key_length;
     uint8_t _flags;
+    ColumnUID _uid = -1;
```
Comment (chaoyli) [inline be/src/column/field.h:186]: What's the usage of `_uid`? Is it determined in FE?

Comment (luohaha) [reply be/src/column/field.h:186]: Use `uid` in DeltaColumnGroup to record which column been updated. This `uid` is came from `TabletColumn`.

File: be/src/storage/olap_common.h (hunk 3 above)
Comment (luohaha) [reply be/src/storage/olap_common.h:273]: invalid column uid is `-1`, same as `TabletColumn`.

File: be/src/column/field.h (hunk 1 above)
Comment (luohaha) [reply be/src/column/field.h]: updated.

**Review by chaoyli — DISMISSED** [review]

**Review by decster — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 81 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadParam.java|0|4|00.00%|[26, 75, 184, 185]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|5|19|26.32%|[323, 324, 325, 326, 327, 328, 329, 430, 472, 473, 474, 475, 476, 477]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|1|2|50.00%|[544]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|2|4|50.00%|[208, 209]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|2|4|50.00%|[89, 117]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|2|3|66.67%|[323]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|5|7|71.43%|[247, 249]|
|:large_blue_circle:|com/starrocks/sql/ast/UpdateStmt.java|9|10|90.00%|[63]|
|:large_blue_circle:|com/starrocks/planner/StreamLoadPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|18|18|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/LoadStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 81 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadParam.java|0|4|00.00%|[26, 75, 184, 185]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|5|19|26.32%|[323, 324, 325, 326, 327, 328, 329, 430, 472, 473, 474, 475, 476, 477]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|1|2|50.00%|[544]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|2|4|50.00%|[208, 209]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|2|4|50.00%|[89, 117]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|2|3|66.67%|[323]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|5|7|71.43%|[247, 249]|
|:large_blue_circle:|com/starrocks/sql/ast/UpdateStmt.java|9|10|90.00%|[63]|
|:large_blue_circle:|com/starrocks/planner/StreamLoadPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|18|18|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/LoadStmt.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 54 / 81 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadParam.java|0|4|00.00%|[26, 75, 184, 185]|
|:large_blue_circle:|com/starrocks/load/streamload/StreamLoadInfo.java|5|19|26.32%|[323, 324, 325, 326, 327, 328, 329, 430, 472, 473, 474, 475, 476, 477]|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadJob.java|1|2|50.00%|[544]|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|2|4|50.00%|[208, 209]|
|:large_blue_circle:|com/starrocks/sql/UpdatePlanner.java|2|4|50.00%|[89, 117]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadJob.java|2|3|66.67%|[323]|
|:large_blue_circle:|com/starrocks/load/loadv2/BrokerLoadJob.java|5|7|71.43%|[247, 249]|
|:large_blue_circle:|com/starrocks/sql/ast/UpdateStmt.java|9|10|90.00%|[63]|
|:large_blue_circle:|com/starrocks/planner/StreamLoadPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateRoutineLoadStmt.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadLoadingTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/UpdateAnalyzer.java|18|18|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/LoadStmt.java|1|1|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Pull request merged**